lib/core/wallet_connect/wallet_connect_service.dart

```diff
@@ -0,0 +1,538 @@
+                  : [],
+              events: requiredEvents.isNotEmpty
+                  ? <String>{...requiredEvents, ...optionalEvents}.toList()
+                  : [],
+            )
+          },
+        );
+
+        log(approval.toJson().toString());
+
+        _walletConnectV2Plugin.approveSession(approval: approval);
+      } catch (e) {
+        log('Approve session error: ${e.toString()}');
+        _bottomSheetHandler.queueBottomSheet(
+          isModalDismissible: true,
+          widget:
+              BottomSheetMessageDisplayWidget(errorText: 'Approve session error: ${e.toString()}'),
+        );
+      }
+    } else {
+      try {
+        _walletConnectV2Plugin.rejectSession(proposalId: proposal.id);
+      } catch (e) {
+        log('Reject session error: ${e.toString()}');
+        _bottomSheetHandler.queueBottomSheet(
+          isModalDismissible: true,
+          widget:
+              BottomSheetMessageDisplayWidget(errorText: 'Reject session error: ${e.toString()}'),
+        );
+      }
+    }
+  }
+
+  Future<void> onSessionSettleEvent(Session session) async {
+    await _refreshSessions();
+    _dappTopic = session.topic;
+    _setDappTopic(_dappTopic!);
+  }
+
+  Future<void> onSessionRejectionEvent(String topic) async {
+    await _refreshSessions();
+  }
+
+  Future<void> onSessionResponseEvent(SessionResponse response) async {
+    log('${response.results is String ? 'Signature' : 'Error'}: ${response.results}');
+    _bottomSheetHandler.queueBottomSheet(
+      isModalDismissible: true,
+      widget: BottomSheetMessageDisplayWidget(
+          errorText: '${response.results is String ? 'Signature' : 'Error'}: ${response.results}'),
+    );
+  }
+
+  void onSessionUpdateEvent(String _) {
+    _refreshSessions();
+  }
+
+  void onSessionDeleteEvent(String _) {
+    _refreshSessions();
+  }
+
+  void onEventError(String code, String message) {
+    log('code: $code | message: $message');
+
+    final bool isInternetConnectionError =
+        code == 'init_core_error' && message.contains('please check your Internet connection');
+
+    final bool isAlreadyInitializedError =
+        code == 'init_sign_error' && message.contains('SignClient already initialized');
+
+    if (isInternetConnectionError || isAlreadyInitializedError) return;
+
+    _bottomSheetHandler.queueBottomSheet(
+      isModalDismissible: true,
+      widget: BottomSheetMessageDisplayWidget(errorText: 'code: $code | message: $message'),
+    );
+  }
+
+//! Subscription related methods
+  @action
+  Future<void> eSignTransactionEvent(SessionRequest request) async {
+    log('Received an ethSignTransaction event\n Request: ${request.toString()}');
+    try {
+      final object = request.params.first as Map<String, dynamic>;
+
+      final ethTransaction = WCEthereumTransaction.fromJson(object);
+
+      final client =
+          Web3Client('https://mainnet.infura.io/v3/51716d2096df4e73bec298680a51f0c5', Client());
```

Comment:
we can revert back to using the current node instead of infura
or did the public nodes cause any issues with wallet connect?

Reply:
Reverted it, waiting on more test eth to test it but it should not give us any issues.